### ISSUE TITLE
bug in svg lossless optimization

### ISSUE THREAD

**Issue by jospoortvliet** [dabutvin/Imgbot#603]
Hi,
Got an image where things go wrong:
https://github.com/nextcloud/nextcloud.com/pull/1248/files#diff-fd432c615a1a70b26b64e3e187a74a8e
https://github.com/nextcloud/nextcloud.com/pull/1248/files#diff-a4c83a7c5d4aa2a21624a18c168bfbab

Anything I can do to help/fix it?

**Comment by dlemstra** [comment]
Can you provide more details on which images are broken?

**Comment by jospoortvliet** [comment]
I think the links above show the images directly, if you click 'display rich diff' - it is the bottom two in the list.

![Screenshot_20200312_080307](https://user-images.githubusercontent.com/551757/76495833-f420f180-6437-11ea-9627-82d331ea6656.png)

**Comment by dabutvin** [comment]
thanks for reporting @jospoortvliet !
We are using [svgo](https://github.com/svg/svgo) for optimizing svgs with these plugins enabled for lossless: https://github.com/dabutvin/Imgbot/blob/master/CompressImagesFunction/Compressors/SvgoCompress.cs#L10-L25

I just checked with the default svgo installation on my laptop with your image and got no visible issue, so it must be something with our configuration. We should go through and track down which plugin is causing the issue and disable it for now as well as report it to the svgo project

**Comment by dabutvin** [comment]
I tracked this down to the `convertStyleToAttrs` plugin in svgo.

Looking at the original nextcloud-hub-logos.svg
Those elements that changed colors have both an inline style and a class on them

one element for example:
```xml
<path style="fill:#ffffff" class="st2" d="M 13.8,15.8 1,9.6 C -0.2,9.1 -0.2,8.3 1,7.8 L 13.8,1.7 c 1.2,-0.5 2.8,-0.5 4,0 l 12.8,6.1 c 1.2,0.5 1.2,1.3 0,1.8 l -12.8,6.1 c -1.1,0.5 -2.9,0.5 -4,0.1 z"/>
```

and the class `st2` is declared in a css block

```css
.st2 {
    fill: #0082c9;
}
```

In this case the inline style attribute wins over the css class rule

When we run the `convertStyleToAttrs` plugin, it converts the element to

```xml
<path fill="#ffffff" class="st2" d="M 13.8,15.8 1,9.6 C -0.2,9.1 -0.2,8.3 1,7.8 L 13.8,1.7 c 1.2,-0.5 2.8,-0.5 4,0 l 12.8,6.1 c 1.2,0.5 1.2,1.3 0,1.8 l -12.8,6.1 c -1.1,0.5 -2.9,0.5 -4,0.1 z"/>
```

replacing the inline style attribute with a fill attribute
The fill rule in the css block associated with the class `st2` wins out over the fill attribute and they change colors.

--

I found an issue logged against the plugin for the same situation in the github repo https://github.com/svg/svgo/issues/1040 

for now I think it would be best for us to disable this plugin until that issue can be resolved.
